File: app/src/main/java/com/mariotti/developer/futureclock/util/AlarmUtil.java
package com.mariotti.developer.futureclock.util;

import android.util.Log;

import com.mariotti.developer.futureclock.models.Alarm;
import com.mariotti.developer.futureclock.models.WeekDay;

public class AlarmUtil {
    private static final String TAG = "AlarmUtil";

    public static String getShortDaysString(Alarm alarm) {
        String result = "NoDay";
        if (alarm != null) {
            int[] days = alarm.getDays();

            if (days.length > 0) {
                result = "";
                for (int i = 0; i < days.length; i++) {
                    result += WeekDay.getShortName(days[i]);
                    if (i < (days.length - 1)) {
                        result += ", ";
                    }
                }
            }
        }

        return result;
    }

    public static String getHourAndMinuteAsString(int hour, int minute) {
        String hourString = (hour < 10) ? "0" + hour : Integer.toString(hour);
        String minuteString = (minute < 10) ? "0" + minute : Integer.toString(minute);
        return hourString + ":" + minuteString;
    }

    public static int[] addDay(int days[], int day) {
        int[] newDays = new int[days.length + 1];
        for (int i = 0; i < days.length; i++) {
            newDays[i] = days[i];
        }
        newDays[days.length] = day;

        Log.d(TAG, "Added day " + WeekDay.getName(day));

        return newDays;
    }

    public static int[] removeDay(int days[], int day) {
        for (int i = 0; i < days.length; i++) {
            if (days[i] == day) {
                days[i] = -1;
                break;
            }
        }
        return WeekDay.reorderDays(days);
    }

    public static boolean hasDay(int days[], int day) {
        if (days.length != 0) {
            for (int dayToCompare : days) {
                if (dayToCompare == day) {
                    return true;
                }
            }
        }

        return false;
    }
}

File: app/src/main/java/com/mariotti/developer/futureclock/util/AlarmUtil.kt
@file:JvmName("AlarmUtil")

package com.mariotti.developer.futureclock.util

import com.mariotti.developer.futureclock.models.Alarm
import com.mariotti.developer.futureclock.models.WeekDay
import java.util.*

fun getDefaultAlarmInstance(): Alarm =
        Alarm(UUID.randomUUID(), Calendar.getInstance(), intArrayOf(), false)

fun getShortDaysString(alarm: Alarm): String {
    var result = StringBuilder()
    val days = alarm.days

    if (days.size > 0) {
        val maxIndex = days.lastIndex
        days.forEachIndexed {
            index, day ->
            result.append(WeekDay.getShortName(day))
            if (index < maxIndex) result.append(", ")
        }
    } else {
        result.append(WeekDay.getShortName(-1))
    }

    return result.toString()
}

fun getHourAndMinuteAsString(hour: Int, minute: Int): String {
    val hourString = if (hour < 10) "0$hour" else Integer.toString(hour)
    val minuteString = if (minute < 10) "0$minute" else Integer.toString(minute)
    return "$hourString:$minuteString"
}

fun addDay(days: IntArray, day: Int): IntArray {
    val newDays = days.toSortedSet()
    newDays.add(day)

    return newDays.toIntArray()
}

fun removeDay(days: IntArray, day: Int): IntArray = WeekDay.reorderDays(days.filter { it != day }.toIntArray())

fun hasDay(days: IntArray, day: Int): Boolean = days.contains(day)